src/util/misc/objectEnlive.ts

```diff
@@ -118,54 +115,39 @@ export const enlivenObjects = <
  * @param {Object} object with properties to enlive ( fill, stroke, clipPath, path )
  * @param {object} [options]
  * @param {AbortSignal} [options.signal] handle aborting, see https://developer.mozilla.org/en-US/docs/Web/API/AbortController/signal
- * @returns {Promise<Record<string, FabricObject | TFiller | null>>} the input object with enlived values
+ * @returns {Promise<Record<string, FabricObject | TFiller | BaseFilter | null>>} the input object with enlived values
  */
 export const enlivenObjectEnlivables = <
-  R = Record<string, FabricObject | TFiller | null>
+  T extends FabricObject | TFiller | BaseFilter,
+  D extends Record<string, Record<string, unknown>> = Record<
+    string,
+    Record<string, unknown>
+  >,
+  R extends Record<keyof D, T> = Record<keyof D, T>
 >(
-  serializedObject: any,
+  serializedObject: D,
   { signal }: Abortable = {}
 ) =>
   new Promise<R>((resolve, reject) => {
-    const instances: (FabricObject | TFiller)[] = [];
+    const instances: T[] = [];
     signal && signal.addEventListener('abort', reject, { once: true });
-    // enlive every possible property
-    const promises = Object.values(serializedObject).map((value: any) => {
-      if (!value) {
-        return value;
-      }
-      // gradient
-      if (value.colorStops) {
-        return new (classRegistry.getClass('gradient'))(value);
-      }
-      // clipPath
-      if (value.type) {
-        return enlivenObjects<FabricObject>([value], { signal }).then(
-          ([enlived]) => {
-            instances.push(enlived);
-            return enlived;
-          }
-        );
-      }
-      // pattern
-      if (value.source) {
-        return classRegistry
-          .getClass('pattern')
-          .fromObject(value, { signal })
-          .then((pattern: Pattern) => {
-            instances.push(pattern);
-            return pattern;
-          });
-      }
-      return value;
+    const promises = Object.values(serializedObject).map(async (value) => {
+      const klass =
+        value &&
+        typeof value === 'object' &&
+        classRegistry.getClass(value, false);
+      if (!klass) return;
```

Comment:
instead of passing all the options and overriding with enlivened ones now this method returns only what was enlivened by the method